### PR TITLE
fix(api): standardize 'areas_ui' to dict format for single area case

### LINKED ISSUE #1507: Impossible to add new area in a layer or district
## Description
We can't add new areas on a layer or a district after we have deleted some old areas whom been there.

## Steps to Reproduce
- create a study with 4 zones
- add a layer
- add zone 2 and 4 to this layer
- delete all areas of the study
- create 6 new areas with new names
- add 2 of these new areas in the layer or in the "All areas" district
- A 404 error appears with the message "zones not found" with the zone names of those that have been deleted

Temporary fix, you have to delete the layer or district and create a new one to be able to add the new areas. 

## Expected Behavior
There should be no errors when adding new areas to the layer or district.

## Screenshots
image

## Environment
- [ ] Production
- [x] Staging

### PR DESCRIPTION
**Pull request by hdinia**
fix #1507

### CODE REVIEW

**Comment by ghost** [comment]
### Bug Fixes

*  **api:** standardize 'areas_ui' to dict format for single area case ([cb5b1ea](https://github.com/AntaresSimulatorTeam/AntaREST/commit/cb5b1eab164f07b3bb86e13d0b598de15b29b483))


### Code Refactoring

* remove `.keys()` which is superfluous ([fff4e82](https://github.com/AntaresSimulatorTeam/AntaREST/commit/fff4e82ae860b503f40d3de3a13521f23ae5233b))


### Contributors

<a href="https://github.com/hdinia">hdinia</a>, <a href="https://github.com/laurent-laporte-pro">laurent-laporte-pro</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

File: antarest/study/business/area_management.py [hunk 1]
```diff
@@ -201,10 +201,14 @@ def update_layer_areas(
         layers = file_study.tree.get(["layers", "layers", "layers"])
         if layer_id not in [str(layer) for layer in list(layers.keys())]:
             raise LayerNotFound
-
         areas_ui = file_study.tree.get(
             ["input", "areas", ",".join(file_study.config.areas.keys()), "ui"]
```
Comment (laurent-laporte-pro) [inline antarest/study/business/area_management.py]: `.keys()` est superflu

File: antarest/study/business/area_management.py [hunk 2]
```diff
@@ -201,10 +201,14 @@ def update_layer_areas(
...
         )
+        # standardizes 'areas_ui' to a dictionary format even if only one area exists.
+        cfg_areas = list(file_study.config.areas.keys())
+        if len(cfg_areas) == 1:
+            areas_ui = {cfg_areas[0]: areas_ui}
+
```
Comment (laurent-laporte-pro) [inline antarest/study/business/area_management.py]: Une autre façon d'implémenter cela est :

```python
# standardizes 'areas_ui' to a dictionary format even if only one area exists.
if len(file_study.config.areas) == 1:
    area = next(iter(file_study.config.areas))
    areas_ui = {area: areas_ui}
```

NOTE: `.keys()` est superflu

File: antarest/study/business/area_management.py [hunk 3]
```diff
@@ -201,10 +201,14 @@ def update_layer_areas(
...
-
         areas_ui = file_study.tree.get(
-            ["input", "areas", ",".join(file_study.config.areas.keys()), "ui"]
+            ["input", "areas", ",".join(file_study.config.areas), "ui"]
         )
+        # standardizes 'areas_ui' to a dictionary format even if only one area exists.
+        cfg_areas = list(file_study.config.areas)
+        if len(cfg_areas) == 1:
+            areas_ui = {cfg_areas[0]: areas_ui}
+
```
Comment (laurent-laporte-pro) [inline antarest/study/business/area_management.py:211]: On peut aussi implémenter de cette façon :

```python
        # standardizes 'areas_ui' to a dictionary format even if only one area exists.
        if len(file_study.config.areas) == 1:
            area = next(iter(file_study.config.areas))
            areas_ui = {area: areas_ui}
```

**Review by laurent-laporte-pro — APPROVED** [review]

**Pull request merged**